### ISSUE TITLE
Revamp the docs for getting started 

### ISSUE THREAD

**Issue by dustinsoftware** [reactjs/React.NET#1005]
Tracking issue for making the documentation more relevant. 

We've seen a few issues (#980, #1002) that appear to be related to brand new projects attempting to use ES module imports. However, the getting started pages ([here](https://reactjs.net/getting-started/aspnetcore.html), [here](https://reactjs.net/tutorials/aspnetcore.html), and [here](https://reactjs.net/getting-started/usage.html)) do not set new projects up with a compatible configuration.

My recommendation: The happy path for getting new projects started with this library should recommend using a bundler, with as minimal of a configuration as possible to support. Optionally we can link out to sites like [createapp.dev](https://createapp.dev/webpack) to preempt questions about more complex configurations. A few bundlers to investigate as the recommended starting point:

- [create-react-app](https://reactjs.org/docs/create-a-new-react-app.html), which ships its own webpack configuration. This project supports very limited customization, asking devs to either "eject" or to maintain a fork of the scripts repo
- [parcel](https://parceljs.org/recipes.html#react), which has built in defaults for React projects, and can be extended
- [webpack](https://webpack.js.org/), which _requires_ a configuration file to work with `babel-loader`. Somewhat ironically, if you google "webpack react", [our webpack bundling page](https://reactjs.net/bundling/webpack.html) is the third result

We probably could add additional documentation for these three bundlers, but the getting started / tutorial should add as little friction as possible for new projects by recommending only one bundler.

Bonus points: This library should throw an error if attempting to use either `require` or `import` in server-side JS, and point to the bundler docs.

Fixes #816

**Comment by dustinsoftware** [comment]
Looked at create-react-app; it provides the nicest out of the box experience but unfortunately it ships error helpers in dev mode that assume access to `window` and `document`.

A fork or craco plugin that creates a server bundle without the error helpers, I am unsure of the long term maintenance consequences however as both would inevitably need patching over time.

No matter what build tooling we recommend, some amount of configuration will be required to get server-render working. 

The `dotnet` CLI [supports templates now](https://docs.microsoft.com/en-us/dotnet/core/tools/custom-templates) that live in nuget packages. `dotnet new React.Template` would be an elegant way to get new projects started, and could be based on the tutorial.

@Daniel15 thoughts? Have you used dotnet templates before?

**Comment by dustinsoftware** [comment]
Did some more investigation. I think any non-webpack route is going to be too much to maintain for SSR; so this issue should be focused on the first run experience and have a reference webpack (basically copies what the sample uses today).

The existing tutorial for non webpack based apps can probably stay as is; but I’m not sure I would recommend it for anything other than one off web pages. We can probably move that to a separate section, eg “what if I don’t want a module bundler”

**Comment by dustinsoftware** [comment]
5.2.0 was published, creating a new `React.Template` package containing installable templates.

```
dotnet new -i React.Template
dotnet new reactnet-vanilla
dotnet run
```

#1049 added the basic template, and #1050 will add the webpack template.

**Comment by dustinsoftware** [comment]
Calling this done as the `reactnet-webpack` template has shipped in 5.2.1. Readme and documentation pages have been updated to recommend the templates first. At this point we're in a much better position to get new users up and running quickly for netcore apps. netfx apps will continue to use the existing samples; at some point they'll go into maintenance mode much like the Owin and Cassette examples.

**Comment by Daniel15** [comment]
> @Daniel15 thoughts? Have you used dotnet templates before?

I haven't, but it sounds like you figured this out 😃
